File: leetcode/961. N-Repeated Element in Size 2N Array/961-2.cpp
// https://helloacm.com/how-to-find-n-repeated-element-in-size-2n-array/ 
// https://leetcode.com/problems/n-repeated-element-in-size-2n-array/

class Solution {
public:
    int repeatedNTimes(vector<int>& A) {
        int i, j;
        while (A[i = rand() % A.size()] != A[j = rand() % A.size()] || i == j);        
        return A[i];
    }
};

File: leetcode/961. N-Repeated Element in Size 2N Array/961-2.php
// https://helloacm.com/how-to-find-n-repeated-element-in-size-2n-array/ 
// https://leetcode.com/problems/n-repeated-element-in-size-2n-array/

class Solution {
    function repeatedNTimes($A) {
        $s = count($A);
        while (($A[$i = rand() % $s] != $A[$j = rand() % $s]) || $i == $j);
        return $A[$i];
    }
}